app/post/postDetailsDirective.js

```diff
@@ -149,32 +155,42 @@
         };
 
         postDetailsCtrl.getTextNumberLikes = function textNumberLikes(numberLikes) {
-             return numberLikes + ' ' + (numberLikes == 1? 'Curtida' : 'Curtidas');
+             return numberLikes;
+        };
+
+        postDetailsCtrl.showLikes = function showLikes(post){
+            return postDetailsCtrl.actualPost === post.key;
         };
 
         postDetailsCtrl.getLikes = function getLikes(post) {
             var likesUri = post.likes;
-
             postDetailsCtrl.current_post = post.key;
             PostService.getLikes(likesUri).then(function success(response) {
-                postDetailsCtrl.likes = response.data;
+                postDetailsCtrl.likes.set(post.key, response.data);
                 postDetailsCtrl.title = post.title;
-                $mdDialog.show({
-                    controller: 'DialogController',
-                    controllerAs: 'dialogCtrl',
-                    templateUrl: 'post/likes.html',
-                    clickOutsideToClose:true,
-                    locals: {
-                        likes : response.data,
-                        title: post.title
-                    }
-                });
-                post.number_of_likes = _.size(postDetailsCtrl.likes);
+                if(postDetailsCtrl.actualPost == post.key){
+                    postDetailsCtrl.actualPost = null;
+                }
+                else{
+                    postDetailsCtrl.actualPost = post.key;
+                }
+                post.number_of_likes = _.size(postDetailsCtrl.likes.get(post.key));
             }, function error(response) {
                 showToast(response.data.msg);
             });
         };
 
+        postDetailsCtrl.getLikesAfterLikeAction = function getLikesAfterLikeAction(post) {
+            PostService.getLikes(post.likes).then(function success(response) {
+                    postDetailsCtrl.likes.set(post.key, response.data);
+                    if(_.size(postDetailsCtrl.likes.get(post.key)) === 0){
```

Comment:
Esse método não é chamado só depois de curtir não? Pq se for sempre vai ter pelo menos uma curtida não?